python/rmm/rmm.py

```diff
@@ -82,6 +86,10 @@ def reinitialize(
     Use `rmm.get_log_filenames()` to get the log file names
     corresponding to each device.
     """
+    [
+        hook(*args, **kwargs)
+        for (hook, args, kwargs) in reversed(_reinitialize_hooks)
```

Comment:
Slight preference to using `func` throughout (as done below). Though maybe `hook_func` is another option

```suggestion
        func(*args, **kwargs)
        for (func, args, kwargs) in reversed(_reinitialize_hooks)
```

Reply:
Fix'd